packages/google_maps_flutter/google_maps_flutter_web/lib/src/marker.dart

```diff
@@ -6,33 +6,35 @@ part of google_maps_flutter_web;
 
 /// The `MarkerController` class wraps a [gmaps.Marker], how it handles events, and its associated (optional) [gmaps.InfoWindow] widget.
 class MarkerController {
-  gmaps.Marker _marker;
+  gmaps.Marker? _marker;
 
   final bool _consumeTapEvents;
 
-  final gmaps.InfoWindow _infoWindow;
+  final gmaps.InfoWindow? _infoWindow;
 
   bool _infoWindowShown = false;
 
   /// Creates a `MarkerController`, which wraps a [gmaps.Marker] object, its `onTap`/`onDrag` behavior, and its associated [gmaps.InfoWindow].
   MarkerController({
-    @required gmaps.Marker marker,
-    gmaps.InfoWindow infoWindow,
+    required gmaps.Marker marker,
+    gmaps.InfoWindow? infoWindow,
     bool consumeTapEvents = false,
-    LatLngCallback onDragEnd,
-    ui.VoidCallback onTap,
+    LatLngCallback? onDragEnd,
+    ui.VoidCallback? onTap,
   })  : _marker = marker,
         _infoWindow = infoWindow,
         _consumeTapEvents = consumeTapEvents {
     if (onTap != null) {
-      _marker.onClick.listen((event) {
+      _marker!.onClick.listen((event) {
```

Comment:
Maybe this method should use `marker` instead everywhere.

Reply:
Yes, in fact the other geometry controllers use the incoming object, instead of the local version.